external_deps/build.sh

```diff
@@ -674,154 +690,134 @@ build_wipe() {
 
 # Common setup code
 common_setup() {
+	HOST="${2}"
+	"common_setup_${1}"
 	WORK_DIR="${PWD}"
 	SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
 	DOWNLOAD_DIR="${WORK_DIR}/download_cache"
 	PKG_BASEDIR="${PLATFORM}_${DEPS_VERSION}"
 	BUILD_BASEDIR="build-${PKG_BASEDIR}"
 	BUILD_DIR="${WORK_DIR}/${BUILD_BASEDIR}"
 	PREFIX="${BUILD_DIR}/prefix"
-	export PATH="${PATH}:${PREFIX}/bin"
-	export PKG_CONFIG="pkg-config"
-	export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig"
-	export CPPFLAGS="${CPPFLAGS:-} -I${PREFIX}/include"
-	export LDFLAGS="${LDFLAGS:-} -L${PREFIX}/lib"
+	PATH="${PATH}:${PREFIX}/bin"
+	PKG_CONFIG="pkg-config"
+	PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig"
+	CPPFLAGS+=" -I${PREFIX}/include"
+	LDFLAGS+=" -L${PREFIX}/lib"
+	case "${PLATFORM}" in
+	*-i686)
+		CFLAGS+=" -msse2"
+		CXXFLAGS+=" -msse2"
+		;;
+	esac
 	if [[ "${MSYSTEM:-}" = MINGW* ]]; then
 		# Experimental MSYS2 support. Most packages won't work;
 		# you need to cross-compile from Linux
 		export CMAKE_GENERATOR="MSYS Makefiles"
-		CROSS= # Doing this because there is no ${CROSS}strip
+		export STRIP='strip' # There is no ${HOST}-strip
 	fi
 	mkdir -p "${DOWNLOAD_DIR}"
-	mkdir -p "${PREFIX}"
 	mkdir -p "${PREFIX}/bin"
 	mkdir -p "${PREFIX}/include"
 	mkdir -p "${PREFIX}/lib"
+	export CC CXX LD AR RANLIB PKG_CONFIG PKG_CONFIG_PATH PATH CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
+}
+
+# -D__USE_MINGW_ANSI_STDIO=0 instructs MinGW to *not* use its own implementation
+# of printf and instead use the system one. It's bad when MinGW uses its own
+# implementation because this can cause extra DLL dependencies.
+# The separate implementation exists because Microsoft's implementation at some
+# point did not implement certain printf specifiers, in particular %lld (long long).
+# Lua does use this one, which results in compiler warnings. But this is OK because
+# the Windows build of Lua is only used in developer gamelogic builds, and Microsoft
+# supports %lld since Visual Studio 2013.
+common_setup_msvc() {
+	CONFIGURE_SHARED=(--enable-shared --disable-static)
+	# Libtool bug prevents -static-libgcc from being set in LDFLAGS
+	CC="${HOST}-gcc -static-libgcc"
+	CXX="${HOST}-g++ -static-libgcc"
+	export DLLTOOL="${HOST}-dlltool"
```

Comment:
Does not need to be exported

Reply:
I removed that `DLLTOOL` variable, it is only used once anyway.